src/value_generators/test/DocumentGeneratorTestCases.yml

```diff
@@ -498,4 +498,40 @@ Tests:
     #    - { "int" : { "$numberLong" : "2000000000"}, "string" : "2000000000" }
     - "string": "2"
 
+  # Now moves with time so would always fail.
```

Comment:
Can we mock out time in C++ so this gives a deterministic value? That would need to be in the DocGenYamlTestCaseRunner, but I think catch2 has some ability for monkey patching. Maybe not. Might be time to look at google test and it's mock support https://github.com/google/googletest/blob/8779937dd05016e3a96e477d2bb1f94947486605/googlemock/docs/cook_book.md FYI @rtimmons .

Reply:
This is just an FYI and for future conversation. Not this ticket.

Reply:
When I first added Catch2 the mocking support wasn't as advanced as it is now and I think the underlying classes had to be designed pretty carefully to support mocking. That seems less true now. It's worth considering adopting more deeply.

We do some primitive dummying of clocks for our rate-limiters tests:
https://github.com/mongodb/genny/blob/3f5b992a916cff4a5408d729b0f391f9e8be0a25/src/gennylib/test/GlobalRateLimiter_test.cpp#L61-L79
But I haven't thought deeply if the same approach works well for value-generators.

Reply:
This could also be related to the suggestion to extend the test runner to support some other approximation checked (see the next comment).

But not in this ticket in any case.

Reply:
Could you add follow-up ticket to the comment?